v4

Skn0tt opened the pull request:
Local PR for getting feedback from Netlify - will open a real PR to the Astro folks next week.

Skn0tt committed:
first attempt at new plugin

Skn0tt committed:
some more change

Skn0tt committed:
remove unneeded handholding from edge functions

Skn0tt committed:
some more work

Skn0tt committed:
remove netlifylocals

Skn0tt committed:
embed astro version

Skn0tt committed:
update README a bunch

Skn0tt committed:
remove per-page-functions

Skn0tt committed:
refactor

Skn0tt committed:
write changeset

Skn0tt committed:
redirects

Skn0tt committed:
update caching

Skn0tt committed:
update cache control

Skn0tt commented:
still needs updates to the test files.

ascorbic commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+## On-Demand Builders are no longer supported
+
+On-Demand Builders (ODB) allows SSR-Rendered pages to be cached using a Time to Live (TTL) strategy.
+ODB continues to be supported by Netlify, but has since been superceded by the much more powerful
```
```suggestion
ODB continues to be supported by Netlify, but has since been superseded by the much more powerful
```

ascorbic commented on packages/netlify/README.md:
````diff
@@ -49,81 +48,44 @@ If you prefer to install the adapter manually instead, complete the following tw
-}
+```astro
+---
+const { geo: { city } } = Astro.locals.netlify.context
````
Why not `const { city } = Astro.locals.netlify.context.geo`?

Skn0tt replied:
no strong preference between the two. Side-note: Do you think `Astro.locals.netlify.context` is a bit verbose? Between these, do you have any preference?
- `Astro.locals.netlifyContext`
- `Astro.locals.context`
- `Astro.locals.netlify`

ascorbic replied:
The reason I suggested changing it is because the syntax is more confusing. For the other question, what el;se is `Astro.locals` used for? If `context` is the only other thing we're adding, I'd say just make it `context` at the top level

Skn0tt replied:
It's what Astro uses to make transport data from the middleware into the SSR rendering: https://docs.astro.build/en/reference/api-reference/#astrolocals
So all other properties on there will be user-configured. `Astro.locals.context` is probably the best, yeah

Skn0tt replied:
one reason that `context` might not be ideal is that Astro already has a `context` parameter they're passing into middleware. Accessing Netlify's context would look like this then:
```ts
import { defineMiddleware } from "astro:middleware";

export const onRequest = defineMiddleware(async (context, next) => {
  context.locals.populatedFromMiddleware = "I was populated from a middleware!";

  context.locals.context // <-- might be confusing
  const resp = (await next()) as Response;
  resp.headers.set("x-astro-middleware", "true");
  return resp;
});
```

this feels a little confusing? makes me think we should keep `netlify` in the name in some way

ascorbic commented on packages/netlify/README.md:
```diff
@@ -146,36 +108,30 @@ Once you run `astro build` there will be a `dist/_redirects` file. Netlify will
-The following example sets a revalidation time of 45, causing Netlify to store the rendered HTML for 45 seconds.
+With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grained-cache-control/), Netlify supports
+standard caching headers like `CDN-Cache-Control` or `Vary`.
+Use these to add caching to your SSR Pages:
```
Maybe mention stale-whike-revalidate specifically?

Skn0tt replied:
done in https://github.com/Skn0tt/adapters/pull/1/commits/3bda429b71b0b379dbbad0593bfa4d3d74d0b39b, let me know how you like that

ascorbic commented on packages/netlify/README.md:
```diff
@@ -146,36 +108,30 @@ Once you run `astro build` there will be a `dist/_redirects` file. Netlify will
 [Read the full deployment guide here.](https://docs.astro.build/en/guides/deploy/netlify/)
 
-After [performing a build](https://docs.astro.build/en/guides/deploy/#building-your-site-locally) the `netlify/` folder will contain [Netlify Functions](https://docs.netlify.com/functions/overview/) in the `netlify/functions/` folder.
+After [performing a build](https://docs.astro.build/en/guides/deploy/#building-your-site-locally) the `.netlify/` folder will contain [Netlify Functions](https://docs.netlify.com/functions/overview/) in the `.netlify/functions-internal/` and [Netlify Edge Functions](https://docs.netlify.com/edge-functions/overview/) in the `.netlify/edge-functions/` folder.
 
 Now you can deploy. Install the [Netlify CLI](https://docs.netlify.com/cli/get-started/) and run:
 
```
This is confusing, because it tells you to perform a build and then says you can deploy, but the dploy command includes `--build`

Skn0tt replied:
done in 95fad12da6ad0cb73ab3eaed7da7f857dcdeefcc

ascorbic commented on packages/netlify/README.md:
````diff
@@ -185,78 +141,7 @@ netlify deploy --build
-  });
-}
-```
+TODO: update astro template after this is released
````
Probably don't want this in the PR!

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+			`
+				import ssrRoute from './entry.mjs';
+				export default ssrRoute;
+				export const config = { name: "Astro SSR", generator: "@astrojs/netlify@${packageVersion}", path: "/*", preferStatic: true };
```
Does preferStatic work for edge functions?

ascorbic replied:
Also can we exclude any paths here?

Skn0tt replied:
doesn't work for edge functions, but this is a netlify function. because we have `preferStatic`, I don't think we need to exclude anything in here - this will be the ultimate fallback, and render a 404 page if there's no matching page.

ascorbic replied:
I was confused by the fact that we're importing types from the wrong place in the function template

Skn0tt replied:
ahh good catch! updated in https://github.com/Skn0tt/adapters/pull/1/commits/18145ca2189fe4947170e3c48e50249908e8e1e1

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+		await build({
+			entryPoints: [fileURLToPath(new URL('./entry.mjs', middlewareOutputDir()))],
+			target: 'es2022',
+			platform: 'node',
```
Why is the platform node?

ascorbic replied:
I think this should be "neutral"

Skn0tt replied:
good catch, updated in https://github.com/Skn0tt/adapters/pull/1/commits/78cfefccb877856c445ab9a4e03ad95036b1c887

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+
+				outDir = new URL('./dist/', rootDir);
+
+				const isRunningInNetlifyDev = process.env.NETLIFY_DEV === 'true'; // TODO
```
What's the TODO?

Skn0tt replied:
not sure if `NETLIFY_DEV` actually exists, this is more of a placeholder for me. gonna address that today

Skn0tt replied:
ha, NETLIFY_DEV isn't a thing, but NETLIFY_LOCAL exists: https://github.com/netlify/build/blob/5eaf9de39853307607b8882ba7a47aef9aa083bf/packages/config/src/env/main.js#L102

Skn0tt replied:
https://github.com/netlify/build/blob/5eaf9de39853307607b8882ba7a47aef9aa083bf/packages/config/src/env/main.js#L188-L191

Skn0tt replied:
and NETLIFY_DEV is also a thing: https://github.com/netlify/cli/blob/5eb2133a277191c68e6d2129f0ebe6d5d85a9402/tests/integration/commands/dev/dev.config.test.js#L128-L143

Skn0tt replied:
updated this in https://github.com/Skn0tt/adapters/pull/1/commits/a95ba6617d13395cf266eff5ca0c6a25fa21e27e. essentially, I want to make sure that we only inject the image CDN handler when there's some netlify thingy running that understands it. on `astro dev`, it should use the normal Astro image handler.

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+
+				const isRunningInNetlifyDev = process.env.NETLIFY_DEV === 'true'; // TODO
+
+				// todo: outut config.image.remotePatterns and config.image.domains into netlify.toml
```
I'd _not_ plan to do this. This should instead rely on the hypotethical new API for frameworks to do this. Mutating the toml is nasty. Instead I would document the fact that they need to add the domains themselves.

Skn0tt replied:
makes sense! added a callout in README and check for this instead in https://github.com/Skn0tt/adapters/pull/1/commits/628248cd0b22952b4808b54149e3adb8fdd35b2f

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+					},
+					image: {
+						service: {
+							entrypoint: !isRunningInNetlifyDev ? '@astrojs/netlify/image-service.js' : undefined,
```
nit: invert this ternary

Skn0tt replied:
nit addressed https://github.com/Skn0tt/adapters/pull/1/commits/854b8251eb15f337049a9f2843391c561f776ac3

ascorbic commented on packages/netlify/src/index.ts:
```diff
@@ -1,2 +1,182 @@
+				if (config.output === 'static') {
+					// eslint-disable-next-line no-console
+					console.warn(
+						`[@astrojs/netlify] \`output: "server"\` or \`output: "hybrid"\` is required to use this adapter.`
```
Is this true if we also use it for generating redirects for static sites?

Skn0tt replied:
fair point, removed it https://github.com/Skn0tt/adapters/pull/1/commits/aa54d88a348fc367d57729236789eea1878ad8bc

Skn0tt commented:
some more TODOs:

- [x] write about image CDN
- [x] test out ntl dev, ntl serve, astro dev

ascorbic commented on packages/netlify/src/ssr-function.ts:
```diff
@@ -0,0 +1,40 @@
+import type { Context } from '@netlify/edge-functions';
+import type { SSRManifest } from 'astro';
+import { App } from 'astro/app';
+import { applyPolyfills } from 'astro/app/node';
```
Why are we applying node polyfills? It doesn't look like they're ones we'd need: 

https://github.com/withastro/astro/blob/main/packages/astro/src/core/polyfill.ts

Skn0tt replied:
this is a Netlify Function that will run on Node v18, so i'm pretty sure we need them

ascorbic replied:
Line 1 says otherwise

Skn0tt replied:
assuming this was the same confusion as above - it's a Lambda Function, i've updated the types import in https://github.com/Skn0tt/adapters/pull/1/commits/18145ca2189fe4947170e3c48e50249908e8e1e1

ascorbic reviewed (approved):
Looking loads better! Great to see this getting some love

ascorbic reviewed (changes requested):
Didn't mean to approve this! There are a couple of changes needed

Skn0tt committed:
Update .changeset/ten-cougars-pretend.md

Co-authored-by: Matt Kane <m@mk.gg>

Skn0tt committed:
update docs

Skn0tt committed:
remove --build

Skn0tt committed:
remove static warning

Skn0tt committed:
nit

Skn0tt committed:
fix: update netlify detection

ascorbic commented on packages/netlify/README.md:
```diff
@@ -146,117 +108,41 @@ Once you run `astro build` there will be a `dist/_redirects` file. Netlify will
-The following example sets a revalidation time of 45, causing Netlify to store the rendered HTML for 45 seconds.
+With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grained-cache-control/), Netlify supports
+standard caching headers like `CDN-Cache-Control` or `Vary`.
+This allows you to implement e.g. Time to live (TTL) or [Stale while revalidate (SWR)](https://docs.netlify.com/platform/caching/#stale-while-revalidate-directive) caching.
```
```suggestion
This allows you to implement e.g. time to live (TTL) or [stale while revalidate (SWR)](https://docs.netlify.com/platform/caching/#stale-while-revalidate-directive) caching.
```

Skn0tt committed:
write about image cdn

Skn0tt committed:
import right types

Skn0tt committed:
Update packages/netlify/README.md

Co-authored-by: Matt Kane <m@mk.gg>

Skn0tt committed:
compile edge middleware with neutral platform

Skn0tt committed:
remove todos

Skn0tt committed:
remove `[astro] update /.netlify/state.json` log lines from local dev

Skn0tt commented:
while looking at `ntl serve` and `ntl dev`, I noticed that vite was watching the `.netlify` directory. disabled that in 7b3f4bb169bdb91867b85e97869b49b5232fd8ea.

there were also some bugs with `ntl serve`, addressed in https://github.com/netlify/cli/pull/6210.

also, the local images didn't seem to work properly - looking at that now.

Skn0tt commented:
this will also need https://github.com/netlify/cli/pull/6211, so `preferStatic` works locally for `ntl serve`

mingjunlu committed:
fix(netlify): broken links in docs (#75)

Skn0tt committed:
remove directories even when they have content

Skn0tt committed:
set "alt" attributes etc on images

jacklyn-net commented on packages/netlify/README.md:
```diff
@@ -49,81 +48,50 @@ If you prefer to install the adapter manually instead, complete the following tw
-When deploying to Netlify Functions, you can choose to use an Edge Function to run your Astro middleware.
-
-To enable this, set the `edgeMiddleware` config option to `true`:
+Netlify Edge Functions provide a [context object](https://docs.netlify.com/edge-functions/api/#netlify-specific-context-object) including metadata about the request, such as a user’s IP, geolocation data, and cookies.
```
[sand] For clarity

```suggestion
Netlify Edge Functions provide a [context object](https://docs.netlify.com/edge-functions/api/#netlify-specific-context-object) that includes metadata about the request such as a user’s IP, geolocation data, and cookies.
```

jacklyn-net commented on packages/netlify/README.md:
```diff
@@ -49,81 +48,50 @@ If you prefer to install the adapter manually instead, complete the following tw
-  // astro.config.mjs
-  import { defineConfig } from 'astro/config';
-  import netlify from '@astrojs/netlify/functions';
+This can be accessed via the `Astro.locals.netlify.context` object:
```
[pebble] Although this is a readme, applying our docs style guide rule here to [avoid the word `via`](https://netlify-handbook.netlify.app/product/documentation/docs-style-guide/terminology/#via)

```suggestion
This can be accessed through the `Astro.locals.netlify.context` object:
```

jacklyn-net commented on packages/netlify/README.md:
````diff
@@ -49,81 +48,50 @@ If you prefer to install the adapter manually instead, complete the following tw
-  }),
-});
-```
+This adapter integrates your site with [Netlify Image CDN](https://docs.netlify.com/image-cdn/), transforming images on-the-fly without impacting build times.
+It's implemented using an [Astro Image Service](https://docs.astro.build/en/reference/image-service-reference/) under the hood.
+Note that this adapter does not support `image.domains` and `image.remotePatterns` config properties in your Astro config - use the `remote_images` field in `netlify.toml` instead: [Netlify Image CDN - Remote Path](https://docs.netlify.com/image-cdn/overview/#remote-path).
````
[dust] Could put this note on its own line (create a new paragraph) to make sure it's easy to see? Right now when it's all one paragraph it blends together a bit. You could also use the `**Note**` formatting.

```suggestion
This adapter integrates your site with [Netlify Image CDN](https://docs.netlify.com/image-cdn/), transforming images on-the-fly without impacting build times.
It's implemented using an [Astro Image Service](https://docs.astro.build/en/reference/image-service-reference/) under the hood.

> **Note**
> This adapter does not support `image.domains` and `image.remotePatterns` config properties in your Astro config - use the `remote_images` field in `netlify.toml` instead: [Netlify Image CDN - Remote Path](https://docs.netlify.com/image-cdn/overview/#remote-path).
```

Skn0tt replied:
good idea! looks a bit like the "warning" callouts we have in our docs :D

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+You can use this to implement customization or authorization logic!
+
+**Action Required:**
+If your site is using Middleware, and you had `edgeMiddleware` enabled, nothing changes for you - feel free to remove the config field, though.
```
[pebble] Maybe something more direct for the last part of the sentence?

```suggestion
If your site is using Middleware, and you had `edgeMiddleware` enabled, nothing changes for you. You are free to remove the config field, as it is no longer necessary.
```

Note that I'm assuming the config field is no longer necessary based on the way this is written?

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+
+**Action Required:**
+If your site is using Middleware, and you had `edgeMiddleware` enabled, nothing changes for you - feel free to remove the config field, though.
+If your site is using Middleware, and you didn't have `edgeMiddleware` enabled, your Middleware will now be deployed to an Edge Function (probably no change needed) and start running on prerendered pages (check if this requires any changes to your middleware logic).
```
[pebble] I feel like the `probably no change needed` brings up questions so I would get rid of it. I would also move the last part to its own sentence, outside of the parentheses.

```suggestion
If your site is using Middleware, and you didn't have `edgeMiddleware` enabled, your Middleware will now be deployed to an Edge Function and start running on prerendered pages. Check if this requires any changes to your middleware logic.
```

However, if we do say `Check if this requires any changes to your middleware logic` it would be nice to say something as to how they can check it? Or link to another section that has the information on how to check for changes?

Skn0tt replied:
> it would be nice to say something as to how they can check it?

They should look at all prerendered pages, and check if they still work as intended now that the middleware applies to them. Will add another line about that 👍

Skn0tt replied:
how do you like https://github.com/Skn0tt/adapters/pull/1/commits/f52963b00fcb7d18e5facc073232c3bd0c2f406b?

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+You can use this to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
+
+**Action Required:**
+If you didn't have a `netlify-edge-middleware.ts` or `netlify-edge-middleware.js` file, you're good.
```
[pebble] Maybe it would help to have a little more info at the end? Like "If you don't have either of these files, you can continue" or something similar? (I'm not sure what the logical "next step" would be here.)

I would also move this to be below the 'Remove the file' content.

Skn0tt replied:
addressed with https://github.com/Skn0tt/adapters/pull/1/commits/7d53619e9111e3f85fd646bb04728b140b45ce69

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+
+In v3, you could use `netlify-edge-middleware.ts` to inject data from the Netlify context into your Astro locals.
+In v4, this file is no longer needed because the Netlify context is automatically made available via `Astro.locals.netlify.context`.
+You can use this to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
```
[sand] To help add some clarity

```suggestion
You can use this context to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
```

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,85 @@
+## On-Demand Builders are no longer supported
+
+On-Demand Builders (ODB) allows SSR-Rendered pages to be cached using a Time to Live (TTL) strategy.
+ODB continues to be supported by Netlify, but has since been superseded by the much more powerful
```
[sand] Simplifying language

```suggestion
ODB continues to be supported by Netlify, but has since been replaced by the much more powerful
```

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
````diff
@@ -0,0 +1,85 @@
+```
+
+**Action Required:**
+If you weren't using ODBs, you're good.
````
[pebble] Same comments as the last `Action Required` comment. I would change the ending to give some kind of action instead of just "you're good". I would also move the sentence to be below the "remove" option content.

jacklyn-net replied:
As a side note, another option for this section and the previous one would be to remove the "you're good" lines altogether. I noticed the following sections just say to remove the option without a failsafe of "but if you don't have this, you're okay". Maybe we can just do that here?

I assume that users would know whether or not they are using ODBs /middleware, so maybe they don't need these extra notes?

Skn0tt replied:
i've removed the "you're good" lines in https://github.com/Skn0tt/adapters/pull/1/commits/7d53619e9111e3f85fd646bb04728b140b45ce69. agree, they're not necessary!

jacklyn-net reviewed (changes requested):
Have some comments here and there to address small things.

For the README.md, are the sections on Edge Context, Image CDN, etc. supposed to be folded under `Installation`? Are they necessary to get things up and running, or are they more optional? At first look it didn't look like they belonged in that section. 🤔

Skn0tt committed:
Apply suggestions from code review

Co-authored-by: Jacklyn <70537879+jacklyn-net@users.noreply.github.com>

Skn0tt committed:
remove "you're good" lines

Skn0tt commented:
- [x] mention Image Service API in changelog

Skn0tt committed:
mention image cdn in changelog

Skn0tt committed:
nest all features under "Usage"

Skn0tt commented:
> For the README.md, are the sections on Edge Context, Image CDN, etc. supposed to be folded under Installation?

Good catch! They belong more under "Usage". Moved them in https://github.com/Skn0tt/adapters/pull/1/commits/96b9078055b4f640db821f6546d69087d5b0c713

Skn0tt committed:
clarify changes to middleware

ascorbic commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -34,6 +34,12 @@ You can use this context to access information about the user (like geolocation
 
+### Image CDN
+
+v4 of this adapter integrates your Astro site with Netlifys [Image CDN](https://docs.netlify.com/image-cdn/overview/) feature.
```
```suggestion
v4 of this adapter integrates your Astro site with Netlify [Image CDN](https://docs.netlify.com/image-cdn/overview/).
```

Skn0tt committed:
Update .changeset/ten-cougars-pretend.md

Co-authored-by: Matt Kane <m@mk.gg>

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,87 @@
+
+**Action Required:**
+If your site is using Middleware, and you had `edgeMiddleware` enabled, nothing changes for you. You are free to remove the config field, as it is no longer necessary.
+If your site is using Middleware, and you didn't have `edgeMiddleware` enabled, your Middleware will now be deployed to an Edge Function and start running on prerendered pages. Go through your prerendered pages, and check if this requires any changes to the logic in your middleware.
```
[pebbles] Some comma fixes, one small word fix.

```suggestion
If your site is using Middleware and you had `edgeMiddleware` enabled, nothing changes for you. You are free to remove the config field, as it is no longer necessary.
If your site is using Middleware and you didn't have `edgeMiddleware` enabled, your Middleware will now be deployed to an Edge Function and start running on prerendered pages. Go through your prerendered pages to check if this requires any changes to the logic in your middleware.
```

jacklyn-net commented on .changeset/ten-cougars-pretend.md:
```diff
@@ -0,0 +1,87 @@
+You can use this context to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
+
+**Action Required:**
+Remove the `netlify-edge-middleware.ts` or ``netlify-edge-middleware.js` file.
```
[pebble] I think there's an extra backtick here that can be removed

```suggestion
Remove the `netlify-edge-middleware.ts` or `netlify-edge-middleware.js` file.
```

jacklyn-net commented on packages/netlify/README.md:
````diff
@@ -49,81 +48,67 @@ If you prefer to install the adapter manually instead, complete the following tw
 ```
 
-#### Pass edge context to your site
+The [Netlify Blog post on Astro](https://www.netlify.com/blog/how-to-deploy-astro/) and the [Netlify Documentation](https://docs.netlify.com/integrations/frameworks/astro/) provide more information on how to use this integration to deploy to Netlify.
````
[sand] "Documentation" sounded a bit weird, think it flows better to just say "Docs", especially since that's how we refer to the site

```suggestion
The [Netlify Blog post on Astro](https://www.netlify.com/blog/how-to-deploy-astro/) and the [Netlify Docs](https://docs.netlify.com/integrations/frameworks/astro/) provide more information on how to use this integration to deploy to Netlify.
```

jacklyn-net commented on packages/netlify/README.md:
```diff
@@ -146,118 +131,26 @@ Once you run `astro build` there will be a `dist/_redirects` file. Netlify will
-By default, all pages will be rendered on first visit and the rendered result will be reused for every subsequent visit until you redeploy. To set a revalidation time, call the [`runtime.setBuildersTtl(ttl)` local](https://docs.astro.build/en/guides/middleware/#locals) with the duration (in seconds).
-
-The following example sets a revalidation time of 45, causing Netlify to store the rendered HTML for 45 seconds.
+With [fine-grained cache control](https://www.netlify.com/blog/swr-and-fine-grained-cache-control/), Netlify supports
+standard caching headers like `CDN-Cache-Control` or `Vary`.
```
[dust] Spacing looks weird here but I think it's probably just how this file displays? Mentioning it in case it needs to be fixed but I think it should be fine?

Skn0tt replied:
I think it's fine, yeah

jacklyn-net reviewed (approved):
Yay! This looks good. Have a few small things but nothing blocking. 🎊

ematipico committed:
chore: use biome to sort imports (#77)

alexanderniebuhr committed:
feat(cloudflare): new local runtime logic (#61)

Co-authored-by: Dario Piotrowicz <61631103+dario-piotrowicz@users.noreply.github.com>
Co-authored-by: Bjorn Lu <34116392+bluwy@users.noreply.github.com>
Co-authored-by: Emanuele Stoppa <602478+ematipico@users.noreply.github.com>
Co-authored-by: Sarah Rainsberger <5098874+sarah11918@users.noreply.github.com>
Co-authored-by: Hippo <6137925+hippotastic@users.noreply.github.com>

alexanderniebuhr committed:
[ci] format

Skn0tt committed:
Apply suggestions from code review

Co-authored-by: Jacklyn <70537879+jacklyn-net@users.noreply.github.com>

ematipico committed:
fix(netlify): add `static` to export map (#80)

* fix(netlify): add `static` to export map

* Update .changeset/dry-news-laugh.md

Co-authored-by: Alexander Niebuhr <alexander@nbhr.io>

---------

Co-authored-by: Alexander Niebuhr <alexander@nbhr.io>

lilnasy committed:
chore(cloudflare): use astro's vite version for plugin type (#82)

ematipico committed:
chore: format with Biome (#79)

* chore: format with Biome

* Add extension to suggestions

* fix missing script

* ignore unknown

* merge conflicts

ematipico committed:
[ci] format

Skn0tt committed:
Merge remote-tracking branch 'upstream/main' into v4

Skn0tt committed:
fix: update function tests

Skn0tt committed:
fix: remove unused test-edge script

Skn0tt committed:
fix: get all tests to run

Skn0tt committed:
fix: workspace name

Skn0tt committed:
fix: redirects on static pages

Skn0tt committed:
feat: add cacheOnDemandPages

Skn0tt committed:
fix: config is optional

lilnasy committed:
update tests to use fixture utils

lilnasy committed:
update fixture for astro 4

lilnasy committed:
fix failure on windows

Skn0tt committed:
refactor: use pathname instead of replacing path sep

Skn0tt committed:
Revert "refactor: use pathname instead of replacing path sep"

This reverts commit 36435e684c581de4883851891766b32785f7239c.

Skn0tt committed:
chore: indentation

vasfvitor committed:
chore: remove `type EndpointOutput` (#83)

alexanderniebuhr committed:
chore(cloudflare): remove deprecated `build.split` option (#86)

alexanderniebuhr committed:
chore: update to Astro v4 (#87)

alexanderniebuhr committed:
chore: fix changeset config (#88)

alexanderniebuhr committed:
fix: try another changeset config (#90)

alexanderniebuhr committed:
fix: remove mixed changeset (#91)

alexanderniebuhr committed:
fix(cloudflare): fix wasm plugin (#89)

Co-authored-by: lilnasy <69170106+lilnasy@users.noreply.github.com>

github-actions[bot] committed:
[ci] release (#73)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

alexanderniebuhr committed:
chore: revert changeset ignore config (#92)

Skn0tt committed:
Merge remote-tracking branch 'upstream/main' into v4

alexanderniebuhr committed:
revert "[ci] release" (#94)

This reverts commit 4b3b0a80a872d3861aaa6c4138c810193c814842.

alexanderniebuhr committed:
chore: ignore netlify release (#95)

github-actions[bot] committed:
[ci] release (#93)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

Skn0tt committed:
Merge branch 'main' into v4

Skn0tt committed:
chore: comment on caching directive

alexanderniebuhr committed:
fix(netlify): tests (#96)

Co-authored-by: Arsh <69170106+lilnasy@users.noreply.github.com>

github-actions[bot] committed:
[ci] release (#98)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

Jinksi committed:
fix(netlify): typo in `peerDependency` range (#100)

github-actions[bot] committed:
[ci] release (#101)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

Skn0tt committed:
Merge remote-tracking branch 'upstream/main' into v4

Skn0tt committed:
mark image cdn as beta

Skn0tt committed:
add edgeMiddleware option back in

Skn0tt committed:
feat: add edge ssr support

Skn0tt committed:
fix: re-add /static and /functions imports

Skn0tt committed:
Revert "feat: add edge ssr support"

This reverts commit 3a03eb80f4513099278d2be9cbe6a7e234b08e18.

Skn0tt committed:
feat: provide netlify context in local dev

Inspired from https://github.com/withastro/adapters/commit/53775ba518b99ac217c2929e5f288fab9506927b

Co-Authored-By: Arsh <cherrycoloredfunk@icloud.com>

Skn0tt committed:
fix: disable edgeMiddleware by default

Skn0tt committed:
Apply suggestions from code review

Co-authored-by: Sarah Rainsberger <sarah@rainsberger.ca>

Skn0tt committed:
update odb changelog

Skn0tt committed:
say we want you to use the replacement!

Skn0tt committed:
Update .changeset/ten-cougars-pretend.md

Co-authored-by: Sarah Rainsberger <sarah@rainsberger.ca>

Skn0tt committed:
Update packages/netlify/README.md
